source/fuzz/transformation_flatten_conditional_branch.cpp

```diff
@@ -0,0 +1,600 @@
+  // header.
+  auto merge_inst = header_block->GetMergeInst();
+  ir_context->KillInst(branch_instruction);
+  ir_context->KillInst(merge_inst);
+
+  // Add a new, unconditional, branch instruction from the current header to
+  // |after_header|.
+  header_block->AddInstruction(MakeUnique<opt::Instruction>(
+      ir_context, SpvOpBranch, 0, 0,
+      std::initializer_list<opt::Operand>{opt::Operand(
+          spv_operand_type_t::SPV_OPERAND_TYPE_ID, {after_header})}));
+
+  // If there is a true branch, change the branch instruction so that the last
+  // block in the true branch unconditionally branches to the first block in the
+  // false branch (or the convergence block if there is no false branch).
+  if (last_true_block) {
+    last_true_block->terminator()->SetInOperand(0, {first_false_block_id});
+  }
+
+  // Replace all of the current OpPhi instructions in the convergence block with
+  // OpSelect.
+  ir_context->get_instr_block(convergence_block_id)
+      ->ForEachPhiInst([&condition_operand](opt::Instruction* phi_inst) {
+        phi_inst->SetOpcode(SpvOpSelect);
+        std::vector<opt::Operand> operands;
+        operands.emplace_back(condition_operand);
+        // Only consider the operands referring to the instructions ids, as the
+        // block labels are not necessary anymore.
+        for (uint32_t i = 0; i < phi_inst->NumInOperands(); i += 2) {
+          operands.emplace_back(phi_inst->GetInOperand(i));
+        }
+        phi_inst->SetInOperands(std::move(operands));
+      });
+
+  // Invalidate all analyses
+  ir_context->InvalidateAnalysesExceptFor(opt::IRContext::kAnalysisNone);
+}
+
+std::unordered_map<opt::Instruction*, std::vector<uint32_t>>
+TransformationFlattenConditionalBranch::GetInstructionsToFreshIdsMapping(
+    opt::IRContext* ir_context) const {
+  std::unordered_map<opt::Instruction*, std::vector<uint32_t>>
+      instructions_to_fresh_ids;
+  for (auto pair : message_.instruction_to_fresh_ids()) {
+    std::vector<uint32_t> fresh_ids;
+    for (uint32_t id : pair.id()) {
+      fresh_ids.push_back(id);
+    }
+
+    auto instruction =
+        FindInstruction(pair.instruction_descriptor(), ir_context);
+    if (instruction) {
+      instructions_to_fresh_ids.emplace(instruction, fresh_ids);
+    }
+  }
+
+  return instructions_to_fresh_ids;
+}
+
+uint32_t TransformationFlattenConditionalBranch::NumOfFreshIdsNeededByOpcode(
+    SpvOp opcode) const {
+  switch (opcode) {
+    case SpvOpStore:
+      return 2;
+    case SpvOpLoad:
+    case SpvOpFunctionCall:
+      return 5;
+    default:
+      assert(false && "This line should never be reached");
+      return 0;
+  }
+}
+
+opt::BasicBlock*
+TransformationFlattenConditionalBranch::EncloseInstructionInConditional(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    opt::BasicBlock* block, opt::Instruction* instruction,
+    std::vector<uint32_t> fresh_ids, uint32_t condition_id,
```

Comment:
Consider using `const std::vector<uint32_t>&` so that we don't make an unnecessary copy of the argument.

Reply:
Done